src/components/views/rooms/RoomHeader.js

```diff
@@ -217,18 +218,27 @@ module.exports = React.createClass({
                 }
 
                 // XXX: this is a bit inefficient - we could just compare room.name for 'Empty room'...
-                var members = this.props.room.getJoinedMembers();
                 var settingsHint = false;
-                if (members.length === 1 && members[0].userId === MatrixClientPeg.get().credentials.userId) {
-                    var name = this.props.room.currentState.getStateEvents('m.room.name', '');
-                    if (!name || !name.getContent().name) {
-                        settingsHint = true;
+                var members = this.props.room ? this.props.room.getJoinedMembers() : undefined;
```

Comment:
Previously the only case where room wasn't set was if simpleHeader was passed, in which case we never got here.